### PR TITLE
Room notifs in autocomplete & composer

### PR DESCRIPTION
**Pull request by dbkr**
A picture paints a thousand words...

screen shot 2017-11-06 at 15 14 01

We should probably remove the max-width here?

Requires https://github.com/vector-im/riot-web/pull/5531

### CODE REVIEW

File: src/autocomplete/NotifProvider.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+*/
+
+import React from 'react';
+import AutocompleteProvider from './AutocompleteProvider';
+import { _t } from '../languageHandler';
+import MatrixClientPeg from '../MatrixClientPeg';
+import {PillCompletion} from './Components';
+import sdk from '../index';
+
+const AT_ROOM_REGEX = /@\S*/g;
```
Comment (lukebarnard1) [inline src/autocomplete/NotifProvider.js]: I guess this means that users starting with the first part (of any length) of `@room` will be slightly more difficult to mention?

File: src/components/views/rooms/MessageComposerInput.js [hunk 2]
```diff
@@ -58,6 +58,9 @@ const TYPING_USER_TIMEOUT = 10000, TYPING_SERVER_TIMEOUT = 30000;
 
 const ZWS_CODE = 8203;
 const ZWS = String.fromCharCode(ZWS_CODE); // zero width space
+
+const ATROOMPILL_ENTITY_TYPE = 'ATROOMPILL';
```
Comment (lukebarnard1) [inline src/components/views/rooms/MessageComposerInput.js]: Maybe
```js
const ENTITY_TYPES = { 
   AT_ROOM_PILL: 'AT_ROOM_PILL' 
};
```

File: src/autocomplete/NotifProvider.js (hunk 1 above)
Comment (dbkr) [reply src/autocomplete/NotifProvider.js]: Yeah, but not sure what we can do about this short of changing it to *room or something

File: src/components/views/rooms/MessageComposerInput.js (hunk 2 above)
Comment (dbkr) [reply src/components/views/rooms/MessageComposerInput.js]: done

File: src/autocomplete/NotifProvider.js (hunk 1 above)
Comment (lukebarnard1) [reply src/autocomplete/NotifProvider.js]: Well we could make it `/@room/`, mandating an entire match, so you'd have to type the whole shebang and then press tab

Comment (dbkr) [reply src/autocomplete/NotifProvider.js]: That would still cause problems for people like late-noughties hitmakers [@room5](https://www.youtube.com/watch?v=UcHs2f2TSxI).

Comment (lukebarnard1) [reply src/autocomplete/NotifProvider.js]: Fair enough!

**Pull request merged**